applications/endoscopy_tool_tracking_distributed/python/cloud_inference_fragment.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from holoscan.core import Fragment
+from holoscan.operators import FormatConverterOp
+from holoscan.resources import (
+    BlockMemoryPool,
+    CudaStreamPool,
+    MemoryStorageType,
+    UnboundedAllocator,
+)
+
+from holohub.lstm_tensor_rt_inference import LSTMTensorRTInferenceOp
+
+# Enable this line for Yuam capture card
+# from holohub.qcap_source import QCAPSourceOp
+from holohub.tool_tracking_postprocessor import ToolTrackingPostprocessorOp
+
+
+class CloudInferenceFragment(Fragment):
+    def __init__(
+        self,
+        app,
+        name,
+        config_key_name,
+        model_dir,
+        width,
+        height,
+        source_block_size,
+        source_num_blocks,
+    ):
+        super().__init__(app, name)
+        self.config_key_name = config_key_name
+        self.model_dir = model_dir
+        self.width = width
+        self.height = height
+        self.source_block_size = source_block_size
+        self.source_num_blocks = source_num_blocks
+
+    def compose(self):
+        source_pool_kwargs = dict(
+            storage_type=MemoryStorageType.DEVICE,
+            block_size=self.source_block_size,
+            num_blocks=self.source_num_blocks,
+        )
+
+        cuda_stream_pool = CudaStreamPool(
+            self,
+            name="cuda_stream",
+            dev_id=0,
+            stream_flags=0,
+            stream_priority=0,
+            reserved_size=1,
+            max_size=5,
+        )
+
+        format_converter = FormatConverterOp(
+            self,
+            name="format_converter",
+            pool=BlockMemoryPool(self, name="pool", **source_pool_kwargs),
+            cuda_stream_pool=cuda_stream_pool,
+            **self.kwargs(self.config_key_name),
+        )
+
+        lstm_inferer_block_size = 107 * 60 * 7 * 4
+        lstm_inferer_num_blocks = 2 + 5 * 2
```

Comment:
Could you please provide a comment to explain how you derived each of these applications?

Reply:
Not really; I copied these from the original endoscopy app where the original didn't have any explanation. :(